### ISSUE TITLE
sonar

### ISSUE THREAD

**Issue by Harshhhh16** [rizacelik/STM32F411CEU6_INAV_Firmware#51]
hello brother i want to add 
1.  VL53L0X TOF Based LIDAR Laser Distance Sensor in this microcontroller
2.  5v Buzzer
3.  servo gimbal
4.  when i connect ublox 6m gps to 5v its led not blinking and heat up what is problem there ? is it 3.3v ??

**Comment by rizacelik** [comment]
1. VL53L0X ToF Distance Ranging Sensor is not supported. All pins are in use. There is no extra pin.
2. For 5v Buzzer, you can connect it instead of CURRENT_METER_ADC in INAV 7.1.0 version. So PB1 pin.
3. You can configure servo gimbal support from Mixer settings.
4. You may have connected the + to - polarities incorrectly on Ublox GPS. Normally it is not possible to warm up. Double check and be careful. Do not burn GPS.

**Comment by Harshhhh16** [comment]
Can i directly connect to Port B1 or any resistor needed between them?
And my gps module is dead or it csn be repaired?

**Comment by rizacelik** [comment]
You can connect a 100ohm resistor to Port B1.
I can't say anything about GPS. You will check it yourself.

**Comment by Harshhhh16** [comment]
Hello
Can i use this type of GPS is there is any problem ?? or what is difference between this module and  without On-board ceramic antenna module ??
GPS NEO-7M Satellite Positioning Module Development Board for Arduino STM32 51
![gps](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/c3a3214c-f1b6-431f-9f7a-eb93f2103be6)

**Comment by rizacelik** [comment]
You can use any GPS module you want. 
There is no difference between them. Only some GPS modules become ready more quickly. So it gets the data sooner. GPS NEO modules initially receive satellite information in a longer time, but there is a slight difference with other modules afterwards.
You can learn the difference between these by examining the datasheet.

**Comment by Harshhhh16** [comment]
What is difference between inav 7.1 vs 6.1 is their any major changes?

**Comment by rizacelik** [comment]
Of course, there are differences between INAV versions. New features are added or removed.
https://github.com/iNavFlight/inav/releases/tag/7.1.0
For a 68K resistor, you can connect more than one resistor in series. Make a calculation to complete 68K by looking at the resistor catalogs in your country.

For any other questions, please refer to the diagram and descriptions. You can find all the information there.

**Comment by Harshhhh16** [comment]
hello can you send quadcopter detail diagram with each and every component on board on latest version of inav ??
because i want to know where to connect black box and some other sensors you provide diagram is okay but i want all component on board embedded diagram....

**Comment by rizacelik** [comment]
I don't have enough time and I try to help you when I have time.
All components you can use with this board are given in the description section. There is no additional information. Please review it thoroughly. How to connect all the components and which pins to use are given. I don't know how else I can help.

**Comment by Harshhhh16** [comment]
Hello
i connect my 7N-0-002 Gps and also all connection are correct and i wait almost 1+hrs for gps signal but still gps not connected with satellite what
![Screenshot 2024-04-22 203534](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/b3f95b7f-6a5a-4bc7-9422-e20130e0f735)
's the problem ?? and i'm also in open space but still not connected :(

**Comment by Harshhhh16** [comment]
Hello
i connect my 7N-0-002 Gps and also all connection are correct and i wait almost 1+hrs for gps signal but still gps not connected with satellite what's the problem ?? and i'm also in open space but still not connected :(





![Screenshot 2024-04-22 203534](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/72096d36-6edd-44ca-8a2d-e0d31e342b1b)
![Screenshot 2024-04-22 204105](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/99a6aa61-19f0-4dbf-9777-efa04ad210a2)

**Comment by rizacelik** [comment]
How did you connect the pins?
GPS pin -> Board pin
Why use INAV 6.0? Have you tried the current version?

**Comment by rizacelik** [comment]
"NEO GPS" modules acquire data after 15 to 25 minutes on first use. Wait for this period. It will get your location information.

**Comment by mursa1en** [comment]
> Hello
> i connect my 7N-0-002 Gps and also all connection are correct and i wait almost 1+hrs for gps signal but still gps not connected with satellite what's the problem ?? and i'm also in open space but still not connected :(
> 
> 
> 
> 
> 
> ![Screenshot 2024-04-22 203534](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/72096d36-6edd-44ca-8a2d-e0d31e342b1b)
> ![Screenshot 2024-04-22 204105](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/99a6aa61-19f0-4dbf-9777-efa04ad210a2)
> 

This GPS module doesn't connect to satellites in indoor conditions. If you're inside your home try to keep the gps near the window if it doesn't solve the issue then try to keep the gps under the open sky.

You can access the inav configurator on the "Speedybee" android app.

**Comment by mursa1en** [comment]
> 1. VL53L0X ToF Distance Ranging Sensor is not supported. All pins are in use. There is no extra pin.
> 2. For 5v Buzzer, you can connect it instead of CURRENT_METER_ADC in INAV 7.1.0 version. So PB1 pin.
> 3. You can configure servo gimbal support from Mixer settings.
> 4. You may have connected the + to - polarities incorrectly on Ublox GPS. Normally it is not possible to warm up. Double check and be careful. Do not burn GPS.

I guess range finders like the vl53l1x/vl53l0x uses i2c to communicate with the flight controller. 

Isn't it possible to utilize the range finder over the existing i2c pins? 

I tried to connect a vl53l1x to the board but it didn't work. Inav (7.0)  didn't detect the module. 
Is the stm32f411ceu6 not capable enough or the firmware doesn't have support for this module?

**Comment by rizacelik** [comment]
Added vl53l0x support. You will use I2C1 pins. You can test it. Write your result here.
[inav_7.1.0_STM32F411CE.zip](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/files/15138123/inav_7.1.0_STM32F411CE.zip)

**Comment by mursa1en** [comment]
Sorry, I accidentally fried my only F411 board with a wrong polarity hookup. So now I can’t test it out. I’ve ordered some more, but it’ll be like two weeks before they show up where I live. and I’ve got the VL53L1X sensor module. I don't have the VL53L0X module.

**Comment by rizacelik** [comment]
vl53l1x/vl53l0x support both. No problem. Support added. You test at the appropriate time.

**Comment by Harshhhh16** [comment]
I want mission planner software for talemetry  which support stm32 f411 microcontroller i tried ardupilot mission planner but its not working :( can you suggest mission planner software ?? or how to connect telemetry to inav ?

**Comment by mursa1en** [comment]
> Added vl53l0x support. You will use I2C1 pins. You can test it. Write your result here.
> [inav_7.1.0_STM32F411CE.zip](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/files/15138123/inav_7.1.0_STM32F411CE.zip)
> 

Vl53l1x successfully working sir 🫡![IMG_20240501_184953.jpg](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/108753302/8f48d00f-ba63-4903-9571-51702114b7e1)

**Comment by Harshhhh16** [comment]
i connect telemetry but there is one issue that is i used inav 6.0 for fc in my case the battery voltage display 9.0v but i have use 3s 2200mah battery and also add 6.6k and 4.3k resistor to the port A1 is there is any additional setup in INAV ??

**Comment by rizacelik** [comment]
INAV 7.1 support the bmi160.
Please see the "README" file.

**Comment by mursa1en** [comment]
> Added vl53l0x support. You will use I2C1 pins. You can test it. Write your result here.
> [inav_7.1.0_STM32F411CE.zip](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/files/15138123/inav_7.1.0_STM32F411CE.zip)
> 

Hello Sir 
The BMP280 barometer is not working over spi/i2c with this firmware but its working fine on others.

**Comment by mursa1en** [comment]
> Added vl53l0x support. You will use I2C1 pins. You can test it. Write your result here.
> [inav_7.1.0_STM32F411CE.zip](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/files/15138123/inav_7.1.0_STM32F411CE.zip)
> 

Hello Sir 
Sorry to bother you again. It seems like that the BMP 280 barometer can't be used in this firmware but it is working fine on the other firmwares ![IMG_20240503_175017.jpg](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/108753302/96c771b3-3020-4ed5-af08-6ca842e32d90)

![IMG_20240503_175222.jpg](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/108753302/9f478f70-9dd9-4435-9b3a-0df3d8d47548)

![IMG_20240503_175222.jpg](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/108753302/d69be4a3-7d61-45ce-af9a-fa7f419441ac)

**Comment by mursa1en** [comment]
I connected it on the i2c pins. Btw i have connected a w25q64 so the nss1 and the the nss2 for the mpu6500 are occupied. Is the nss3 on the pin c14 usable?

**Comment by rizacelik** [comment]
To use I2C, use the following CLI commands.

set baro_bustype = I2C
set baro_spi_device = 0
set baro_i2c_device = 1

**Comment by rizacelik** [comment]
No, PC15 pin for SPI2 is BMP280 barometer NS pin. Use pin PC15 for barometer.

**Comment by mursa1en** [comment]
Im getting these errors 
![IMG_20240503_190229.jpg](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/108753302/2038a847-50e9-4e73-81d8-43b01bfc9bf3)

**Comment by mursa1en** [comment]
Working fine over spi. Thanks for your help Sir

**Comment by 8305-kavin** [comment]
1.could you specify in which pins you have connected the bmp 280 and the sonar ?
2.Cant we use telemetry ?
3.Can you figure out a way to use 3dr telmetry insted of the black box?

**Comment by rizacelik** [comment]
Please read the [README](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware#note-1) file. I gave links to telemetry above in this message. Read them too.
